### ISSUE TITLE
3.5.rc1 Editor window fail to close

### ISSUE THREAD

**Issue by Abdera7mane** [godotengine/godot#61261]
### Godot version

v3.5.rc1.official [e86d840d4]

### System information

Windows 7 & 10, GLSE3

### Issue description

Closing the Editor causes the window to stop responding, this also occurs when Godot switches from the Project Manager to the Project workspace (the project manager window stops responding and fails to close), this wasn't present with **3.4.4.stable** and **3.5.beta5**. Other than that it appears that the engine is functioning as expected so far. The reason to this problem is probably a memory leak on the loaded resources.

here is the output with `--verbose` flags.

```
D:\Programes\Godot>
Godot Engine v3.5.rc1.official.e86d840d4 - https://godotengine.org
Using GLES3 video driver
OpenGL ES 3.0 Renderer: GeForce 9500 GT/PCIe/SSE2
Async. shader compilation: OFF
OpenGL ES 2D Batching: ON
Batching Options:
        max_join_item_commands 16
        colored_vertex_format_threshold 0.25
        batch_buffer_size 16384
        light_scissor_area_threshold 1
        item_reordering_lookahead 4
        light_max_join_items 32
        single_rect_fallback False
        debug_flash False
        diagnose_frame False
WASAPI: wFormatTag = 65534
WASAPI: nChannels = 2
WASAPI: nSamplesPerSec = 48000
WASAPI: nAvgBytesPerSec = 384000
WASAPI: nBlockAlign = 8
WASAPI: wBitsPerSample = 32
WASAPI: cbSize = 22
WASAPI: detected 2 channels
WASAPI: audio buffer frames: 933 calculated latency: 21ms

CORE API HASH: 13016675760375384316
EDITOR API HASH: 559252884276539986
Loading resource: C:/Users/Abdera7mane/AppData/Roaming/Godot/editor_settings-3.tres
EditorSettings: Load OK!
Scanning projects at: D:\Programs\Godot\Projects
Found 132 projects.
EditorSettings: Save OK!
Loaded builtin certs
EditorSettings: Save OK!
WARNING: ObjectDB instances leaked at exit (run with --verbose for details).
     at: cleanup (core/object.cpp:2071)
Leaked instance: Image:8441 - Resource path:
Hint: Leaked instances typically happen when nodes are removed from the scene tree (with `remove_child()`) but not freed (with `free()` or `queue_free()`).
Orphan StringName: Image
StringName: 1 unclaimed string names at exit.
ERROR: There are still MemoryPool allocs in use at exit!
   at: (core/pool_vector.cpp:63)

```

### Steps to reproduce

the shortest way to reproduce:
1. open Godot.
2. wait for the Project Manager to load projects list.
3. close the window.

which provides the previous console output

**Comment by Calinou** [comment]
Can you look into [bisecting](https://docs.godotengine.org/en/stable/community/contributing/bisecting_regressions.html) the exact commit in which the regression started? This will require compiling the engine several times, but it'll greatly speed up troubleshooting.

**Comment by Abdera7mane** [comment]
Took me a while, 2d3f5855c62588c6bd5145aa3b53c3902dbe499f is the first bad commit.

**Comment by akien-mga** [comment]
Thanks for doing that work, it's very useful to identify and fix the regression.

**Comment by bruvzg** [comment]
> Took me a while, https://github.com/godotengine/godot/commit/2d3f5855c62588c6bd5145aa3b53c3902dbe499f is the first bad commit.

Unlike `DisplayServer` on master, `OS` is destroyed too late to clean `icon`. It's probably should be manually removed before `ObjectDB` is cleared. I'll take a look tomorrow.

**Comment by akien-mga** [comment]
Fixed by #61304.
